Add missing sections to the CSS property page template (#22462)
mdn/content: files/en-us/mdn/writing_guidelines/page_structures/page_types/css_property_page_template/index.md

{{MDNSidebar}}

> > **Note:** _Remove this whole explanatory note before publishing._

Reviewer: Minor reword:

```suggestion
> **Note:** _Remove this note block before publishing._
```

One thing that's helpful for me to see what to remove is to have start/end markers like this:

```
<!-- USAGE NOTE START -->
...
<!-- USAGE NOTE END -->
```

But this might be overkill because the note block here is highlighted from the rest of the page anyway.

Author: Agree and agree!

> ```
>
> - **title**

> >   - : Title heading is displayed at the top of the page. The title format is _NameOfTheProperty_.

Reviewer: ```suggestion
>   - : The title heading is displayed at the top of the page. The title format is _NameOfTheProperty_.
```

>     For example, the [`background-color`](/en-US/docs/Web/CSS/background-color) property has a title of _background-color_.
> - **slug**
>   - : The `slug` value is the end of the URL path after `https://developer.mozilla.org/en-US/docs/`. This will be formatted like `Web/CSS/NameOfTheProperty`.
>     For example, the [`background-color`](/en-US/docs/Web/CSS/background-color) property slug is `Web/CSS/background-color`.
> - **page-type**
>   - : The `page-type` key for CSS properties is `css-shorthand-property` for shorthand properties; otherwise it is `css-property`. For example, the `page-type` value for the [animation](/en-US/docs/Web/CSS/animation) property is `css-shorthand-property` but for the the [animation-delay](/en-US/docs/Web/CSS/animation-delay) property, it is `css-property`.
> - **tags**
>
>   - : Always include the following tags: **CSS**, **Reference**, **CSS Property**, _NameOfTheProperty_ (e.g. **background-color**).
>
>     Include the following tags as appropriate:
>
>     - Category of property: **Layout**, **Graphics**, **CSS Background**, **CSS Colors**, _other tags as appropriate_

> >     - Technology status: **Experimental** (if the technology is [experimental](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#experimental)), **Deprecated** (if it is [deprecated](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#deprecated_and_obsolete))

Reviewer: Should we make this shorter? 

```suggestion
>     - Technology status: Can be either [**Experimental**](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#experimental) or [**Deprecated**](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#deprecated_and_obsolete)
```

Also, does capitalization matter?

Author: Shortened 👍 . As far as I've seen, the tags are mentioned with initial caps

>
> ---
>
> **Other macros in the page**
>

> > - Formal definition and Formal syntax sections: The content of the _Formal definition_ and _Formal syntax_ sections is generated in place of the `\{{CSSInfo}}` and `\{{CSSSyntax}}` macros, respectively. For these sections to populate, you must ensure an appropriate entry has been filled in for the property in our [properties.json](https://github.com/mdn/data/blob/main/css/properties.json) data file. See [properties.md](https://github.com/mdn/data/blob/main/css/properties.md) for more information.

Reviewer: ```suggestion
> - Formal definition and Formal syntax sections: The content of the _Formal definition_ and _Formal syntax_ sections is generated in place of the `\{{CSSInfo}}` and `\{{CSSSyntax}}` macros, respectively. For these sections to have data, you must ensure an appropriate entry has been filled in for the property in our [properties.json](https://github.com/mdn/data/blob/main/css/properties.json) data file. See [properties.md](https://github.com/mdn/data/blob/main/css/properties.md) for more information.
```

> > - Specifications and Browser compatibility sections: The toolchain automatically uses the `browser-compat` key-value pair to populate the Specifications and Browser compatibility sections (replacing the `\{{Specifications}}` and `\{{Compat}}` macros in those sections, respectively).

Reviewer: ```suggestion
> - Specifications and Browser compatibility sections: The build tool automatically uses the `browser-compat` key-value pair to insert data into the Specifications and Browser compatibility sections (replacing the `\{{Specifications}}` and `\{{Compat}}` macros in those sections, respectively).
```

>

> >     Note that you may first need to create/update an entry for the property and its specification in our <a href="https://github.com/mdn/browser-compat-data">Browser compat data repo</a>.

Reviewer: Prettier doesn't like the indentation:

```suggestion
>   Note that you may first need to create/update an entry for the property and its specification in our <a href="https://github.com/mdn/browser-compat-data">Browser compat data repo</a>.
```

> >     See our [guide on how to do this](/en-US/docs/MDN/Writing_guidelines/Page_structures/Compatibility_tables).

Reviewer: ```suggestion
>   See our [compatibility tables guide](/en-US/docs/MDN/Writing_guidelines/Page_structures/Compatibility_tables) for information on adding or editing entries.
```

>

> > _Remember to remove this whole explanatory note before publishing._

Reviewer: ```suggestion
> _Remember to remove this note block before publishing._
```

{{SeeCompatTable}}{{deprecated_header}}{{CSSRef}}

In the introductory paragraph, start by naming the property and saying what it does.
This should ideally be one or two short sentences.

## Try it

_This title is auto-generated by the macro `\{{EmbedInteractiveExample}}`._

This section is for interactive example, which gets adding by calling the macro `\{{EmbedInteractiveExample}}`. See the [Interactive example demo](/en-US/docs/MDN/Writing_guidelines/Page_structures/Code_examples#interactive_example_demo) section in our writing guidelines for more information.

## Constituent properties

> This section needs to be added only for shorthand properties like [animation](/en-US/docs/Web/CSS/animation) to list all the related long-hand properties.

Reviewer: ```suggestion
Only add this section for shorthand properties, such as [animation](/en-US/docs/Web/CSS/animation), to list all the related longhand properties.
```

## Syntax

> Include the common use cases as a code block and describe the component subvalues that make up a complete value.

Reviewer: Could we elaborate on "component subvalues that make up a complete value"? Is there an example?

Author: I am as clueless as you - no idea what that means.

```css
/* Insert code block showing common use cases */
/* or categories of values */
```

### Values

Include one term and definition for each subvalue.

- subvalue1

>   - : Include a description of the subvalue, its datatype, and what it represents.

Reviewer: Do we have guidelines for `datatype` v `data type`?

Author: Should be data type! :)

- subvalue2
  - : Include a description of the subvalue, its datatype, and what it represents.

## Description

This is an optional section to include a description of the property and explain how it works. Use this section to explain related terms and add use cases for the property.

## Formal definition

`\{{CSSInfo}}`

> _Remember to remove the codes and backslash in the markdown file when using this macro call._

Reviewer: ```suggestion
_Remember to remove the backticks and backslash in the markdown to use this macro._
```

A code block might also be useful:

````
```
{{CSSInfo}}
```
````

See [common macros](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Macros/Commonly_used_macros#indicating_that_a_feature_is_available_in_web_workers)

Author: A code block looks better. Strangely, the code block works with {{CSSSyntax}} but with {{CSSInfo}}, the macro seems to be functional and trying to fetch data. So I guess a backslash is needed.

## Formal syntax

`\{{CSSSyntax}}`

> _Remember to remove the codes and backslash in the markdown file when using this macro call._

Reviewer: ```suggestion
_Remember to remove the backticks and backslash to use this macro._
```

## Accessibility concerns

> This is an optional section. You can include here any warnings for any accessibility concerns that developers should be aware of while using this property. You can also include if there are workarounds for these accessibility concerns.

Reviewer: ```suggestion
This is an optional section. You can include any warnings here for accessibility concerns that developers should be aware of while using this property. You can also include if there are workarounds for these accessibility concerns.
```

## Specifications

`\{{Specifications}}`

> _Remember to remove the codes and backslash in the markdown file when using this macro call._

Reviewer: ```suggestion
_Remember to remove the backticks and backslash to use this macro._
```

## Browser compatibility

`\{{Compat}}`

> _Remember to remove the codes and backslash in the markdown file when using this macro call._

Reviewer: ```suggestion
_Remember to remove the backticks and backslash to use this macro._
```

## See also

> Include here a list of other properties or guides related to the current property page

Reviewer: ```suggestion
Include guides relating to the current property or a list of other properties here.
```
